### PR TITLE
bugfix: postgres_info.py now shows the info of each specific database

### LINKED ISSUE #172: community.postgresql.postgresql_info is showing the namespaces of "db" variable for every database
##### SUMMARY

let's say I have :

* `db1`, with namespaces `ns1A` and `ns1B`
* `db2` with namespaces `ns2A` and `ns2B`

whenever I use 

```

  - community.postgresql.postgresql_info:
      login_host: 127.0.0.1
      login_user: root
      login_password: super_secret_password
      db: "db1"
    register: db_info

  - debug:
      msg: "{{ db_info }}"

```

it will print `ns1A` and `ns1B` both as the namespaces of both `db1` and `db2`. 
Not only that, `ns2A` and `ns2B` won't be returned at all.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
community.postgresql.postgresql_info

##### ANSIBLE VERSION

``` # ubuntu 20.04

ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/mdiez/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.10 (default, Sep 28 2021, 16:10:42) [GCC 9.3.0]


```

##### COLLECTION VERSION

```master

```

### PR DESCRIPTION
**Pull request by marcosdiez**
##### SUMMARY
fixes #172

The reason `postgres_info.py` was gathering data on every database was because of a bug in `PgDbConn.reconnect()`, which is fixed here.

I also changed the main loop that iterate among all DBs to ignore DBs which it can't access because in AWS RDS not all DBs are accessible.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
postgres_info.py

### CODE REVIEW

**Comment by Andersson007** [comment]
@marcosdiez hi and welcome to the project! Thank you for the bunch of issues and fixes! Could you please:
- Make CI green
- Add a changelog fragment to each PRs https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment
- Add integration tests (where possible). See the content of `tests/integration/target/postgresql_info/tasks`

**Comment by Andersson007** [comment]
@marcosdiez thank you for the PR, could you please:
1. Fix CI, everything should be green (see the "Details" link for details)
2. Be sure the changes is covered with integration tests, see the content of `tests/integration/targets/postgresql_info/tasks`

**Comment by Andersson007** [comment]
closed reopened to re-trigger CI

**Comment by Andersson007** [comment]
@marcosdiez ah, the failings are unrelated (there are AZP issues there), sorry
So please cover the case mentioned in https://github.com/ansible-collections/community.postgresql/issues/172 somewhere within `tests/integration/targets/postgresql_info/tasks` and let us know when it's ready for review

File: changelogs/fragments/173-postgres_info_now_shows_the_info_of_each_specific_database.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- postgres_info - Specific info (namespaces, extensions, languages) of each database was not being shown properly. Instead, the info from the DB that was connected was always being shown.
```
Comment (Andersson007) [inline changelogs/fragments/173-postgres_info_now_shows_the_info_of_each_specific_database.yml:2]:
```suggestion
- postgres_info - Specific info (namespaces, extensions, languages) of each database was not being shown properly. Instead, the info from the DB that was connected was always being shown (https://github.com/ansible-collections/community.postgresql/issues/172).
```

File: plugins/modules/postgresql_info.py [hunk 2]
```diff
@@ -1012,6 +1019,13 @@ def get_db_info(self):
 
         for datname in db_dict:
             self.cursor = self.db_obj.reconnect(datname)
+            if self.cursor is None:
+                # that means we don't have permission to access these database
+                db_dict[datname]['namespaces'] = {}
+                db_dict[datname]['extensions'] = {}
+                db_dict[datname]['languages'] = {}
+                db_dict[datname]['error'] = "could not connect to the database (probably permission issue)"
```
Comment (Andersson007) [inline plugins/modules/postgresql_info.py:1027]:
```suggestion
                db_dict[datname]['error'] = "Could not connect to the database."
```
I suggest avoiding assumptions. Later we could think how to return actual errors.

**Review by Andersson007 — CHANGES_REQUESTED** [review]

**Comment by marcosdiez** [comment]
@Andersson007 hi! Unit tests added. Sorry for the delay, but it took me longer to find out how to run unit tests locally then to actually write them.

By the way, the solution I found is `time ansible-test integration --target docker:ubuntu2004 --docker-terminate never -v postgresql_info` which is taking a little more than a minute on each iteration.

I still have not found a way to use the same container and hence not  have to download postgres every time I want to run unit tests.

Please feel free to core review this PR and request changes as needed.
Thanks!

**Review by Andersson007 — APPROVED** [review]
LGTM

**Comment by Andersson007** [comment]
@marcosdiez thanks for the great contribution!:)

**Comment by Andersson007** [comment]
> By the way, the solution I found is time ansible-test integration --target docker:ubuntu2004 --docker-terminate never -v postgresql_info which is taking a little more than a minute on each iteration.

I just use `ansible-test integration postgresql_info --docker ubuntu2004 -vvv > ~/test.log`, never dug into it deeply. I'll compare your command with this simplest one next time, thank you!

**Pull request merged**